### PR TITLE
[chore] move package tests from contrib repo to this repo

### PR DESCRIPTION
**Pull request by mowies**
### This PR
- moves the package tests from https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/internal/buildscripts/packaging/fpm to this repo
- didn't copy over all of the files from the other repo since they were not needed due to a slightly different testing approach where the packages are built earlier in the pipeline and not right before testing
- fixes https://github.com/open-telemetry/opentelemetry-collector-releases/issues/439
- ~fixes some pipeline `if` conditionals that didn't work correctly~ separate PR: https://github.com/open-telemetry/opentelemetry-collector-releases/pull/610
- ~bumps goreleaser in the `go.mod` file to v2 since that is the version that's used in all pipelines anyways (see e.g. [this goreleaser usage](https://github.com/open-telemetry/opentelemetry-collector-releases/blob/main/.github/workflows/base-ci-goreleaser.yaml#L70))~ separate PR: https://github.com/open-telemetry/opentelemetry-collector-releases/pull/609

#### Testing
I tested the new pipelines on my fork's main branch. Fully working runs with the setup from this PR can be seen here:
[Core pipeline](https://github.com/mowies/opentelemetry-collector-releases/actions/runs/10076035554), [Contrib pipeline](https://github.com/mowies/opentelemetry-collector-releases/actions/runs/10076035572)

#### Notes
- The `package-tests.sh` and `common.sh` together with the test dockerfiles were copied over and slightly adjusted where needed.
- I didn't add a test setup for the .apk package version since that was also not present in the other repo and I didn't want to increase the size of this PR with additional new stuff. That part could be a follow up issue.

#### Testing
I merged this PR into my forked main branch and the pipelines fully work there.
Links to the pipeline runs: [Core pipeline](https://github.com/mowies/opentelemetry-collector-releases/actions/runs/10076035554), [Contrib pipeline]()

#### Follow ups
Remove package tests from contrib repo

### CODE REVIEW

File: .github/workflows/base-ci-goreleaser.yaml [hunk 1]
```diff
@@ -46,7 +46,7 @@ jobs:
           platforms: arm64,ppc64le,linux/arm/v7,s390x
 
       - name: Setup wixl # Required to build MSI packages for Windows
-        if: ${{ matrix.GOOS }} == 'windows' && (${{ matrix.GOARCH }} == '386' || ${{ matrix.GOARCH }} == 'amd64')
```
Comment (mx-psi) [inline .github/workflows/base-ci-goreleaser.yaml:49]: Can we split this fix and the check in base-release into a separate PR?

File: .goreleaser.yaml [hunk 2]
```diff
@@ -1,3 +1,4 @@
+version: 2
```
Comment (mx-psi) [inline .goreleaser.yaml:1]: Can we do the v2 bump on a separate PR?

File: .github/workflows/base-ci-goreleaser.yaml (hunk 1 above)
Comment (mowies) [reply .github/workflows/base-ci-goreleaser.yaml:49]: sure

File: .goreleaser.yaml (hunk 2 above)
Comment (mowies) [reply .goreleaser.yaml:1]: sure

Comment (mowies) [reply .goreleaser.yaml:1]: https://github.com/open-telemetry/opentelemetry-collector-releases/pull/609

File: .github/workflows/base-ci-goreleaser.yaml (hunk 1 above)
Comment (mowies) [reply .github/workflows/base-ci-goreleaser.yaml:49]: https://github.com/open-telemetry/opentelemetry-collector-releases/pull/610

**Comment by mowies** [comment]
@mx-psi I separate out the changes you requested and will rebase this PR once those are merged to bring it up to date again.
Here are the links: https://github.com/open-telemetry/opentelemetry-collector-releases/pull/609, https://github.com/open-telemetry/opentelemetry-collector-releases/pull/610

**Comment by mowies** [comment]
@mx-psi both #609 and #610 are merged now and I rebased this PR.
should be ready for review again :)

File: scripts/package-tests/Dockerfile.test.deb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# A debian11 image with systemd enabled.  Must be run with:
```
Comment (mx-psi) [inline scripts/package-tests/Dockerfile.test.deb:1]: This comment seems wrong. Should this say 
```suggestion
# A debian12 image with systemd enabled.  Must be run with:
```
or maybe
```suggestion
# A debian image with systemd enabled.  Must be run with:
```
?

File: scripts/package-tests/Dockerfile.test.rpm [hunk 4]
```diff
@@ -0,0 +1,9 @@
+# A rockylinux8.5 image with systemd enabled.  Must be run with:
```
Comment (mx-psi) [inline scripts/package-tests/Dockerfile.test.rpm:1]: ditto, 
```suggestion
# A rockylinux9.3 image with systemd enabled.  Must be run with:
```

or maybe
```suggestion
# A rockylinux image with systemd enabled.  Must be run with:
```

File: scripts/package-tests/README.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+Tools:
+
+- [Go](https://go.dev/)
+- [GoReleaser](https://goreleaser.com/)
+- [Podman](https://podman.io/)
+- make
+
+## How to build and test
+
+To build the collector linux packages, a few steps are required:
```
Comment (mx-psi) [inline scripts/package-tests/README.md:14]:
```suggestion
To build the Collector Linux packages, a few steps are required:
```

File: scripts/package-tests/README.md [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+- Run `make generate` to (re-)generate sources and goreleaser files
```
Comment (mx-psi) [inline scripts/package-tests/README.md:16]:
```suggestion
- Run `make generate` to (re-)generate sources and GoReleaser files
```

File: scripts/package-tests/README.md [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+- Go to the distribution folder that you want to build (under the `distributions` folder)
+- Run `goreleaser release --snapshot --clean --skip sbom,sign,archive,docker`
+    - This will build a full release with all architectures and packaging types into the `dist` folder inside your
+      current folder. (We can skip many parts of the release build that we don't need)
```
Comment (mx-psi) [inline scripts/package-tests/README.md:20]: I don't understand the comment about skipping parts. Who is 'we'? How can these parts be skipped? Why are they not needed?

File: scripts/package-tests/README.md [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+    - If you run into `unmarshal` errors, start to remove the parts that goreleaser complains about. This likely happens
+      due to a missing goreleaser pro license and therefore feature that you can't use.
```
Comment (mx-psi) [inline scripts/package-tests/README.md:22]:
```suggestion
    - We use GoReleaser Pro only features in CI. If you want to run this locally and you run into `unmarshal` errors, 
    you may have to remove the parts that goreleaser complains about or use a pro license.
```

File: scripts/package-tests/README.md (hunk 7 above)
Comment (mowies) [reply scripts/package-tests/README.md:20]: we as in the users :)
many of the goreleaser steps can be skipped for the package tests as they are not needed for those.

Comment (mx-psi) [reply scripts/package-tests/README.md:20]: Thanks for clarifying! So this comment is also intended for developers running the tests locally then?

Comment (mowies) [reply scripts/package-tests/README.md]: I updated the wording slightly to hopefully make it a bit clearer

**Review by mx-psi — APPROVED** [review]
Thanks!

**Comment by mx-psi** [comment]
I am going to wait to merge this until after v0.106.1 is released

**Pull request merged**